rsyslog/go/src/github.com/mmexternal/undefined_field/undefined_field.go

```diff
@@ -0,0 +1,239 @@
+package main
+
+import (
+	"encoding/json"
+	"bufio"
+	"strconv"
+	"strings"
+	"fmt"
+	"os"
+)
+
+const (
+	initial_use_undefined              = false
+	initial_undefined_name             = "undefined"
+	initial_undefined_to_string        = false
+	initial_default_keep_fields        = "CEE,time,@timestamp,aushape,ci_job,collectd,docker,fedora-ci,file,foreman,geoip,hostname,ipaddr4,ipaddr6,kubernetes,level,message,namespace_name,namespace_uuid,offset,openstack,ovirt,pid,pipeline_metadata,rsyslog,service,systemd,tags,testcase,tlog,viaq_msg_id"
+	initial_extra_keep_fields          = ""
+	initial_keep_empty_fields          = ""
+	initial_undefined_max_num_fields   = -1
+	initial_undefined_dot_replace_char = "UNUSED"
+	initial_logging_file_path          = "/var/log/rsyslog/rsyslog.log"
+)
+
+var (
+	use_undefined bool
+	keep_fields []string
+	keep_empty_fields []string
+	undefined_name string
+	undefined_to_string bool
+	undefined_dot_replace_char string
+	undefined_max_num_fields int64
+	logfile *os.File
+	noaction          = false
+	replacer          = &strings.Replacer{}
+	basic_keep_fields = "_STREAM_ID,_SYSTEMD_INVOCATION_ID"
+)
+
+func onInit() {
+	// opening the rsyslog log file
+	logging_file_path := initial_logging_file_path
+	if eval := os.Getenv("LOGGING_FILE_PATH"); eval != "" {
+		logging_file_path = eval
+	}
+	if f, err := os.OpenFile(logging_file_path, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644); err == nil {
+		logfile = f
+	} else {
+		panic(err.Error())
+	}
+
+	// initializing
+	use_undefined = initial_use_undefined
+	default_keep_fields := initial_default_keep_fields
+	extra_keep_fields := initial_extra_keep_fields
+	undefined_name = initial_undefined_name
+	tmp_keep_empty_fields := initial_keep_empty_fields
+	undefined_to_string = initial_undefined_to_string
+	undefined_dot_replace_char = initial_undefined_dot_replace_char
+	undefined_max_num_fields = initial_undefined_max_num_fields
+
+	config, err := os.Open("/var/lib/rsyslog.pod/undefined.yaml")
+	if err != nil {
+	  panic(err.Error())
+	}
+	defer config.Close()
+
+	reader := bufio.NewReader(config)
+	scanner := bufio.NewScanner(reader)
+	scanner.Split(bufio.ScanLines)
+	for scanner.Scan() {
```

Comment:
I would encourage you to write out the config file in `yaml` format rather than `key: value` format.  Then you can use the built-in yaml parser rather than having to write your own parser.

Reply:
I see - there is no built-in yaml parser - a quick check of our operator code shows that `"github.com/ghodss/yaml"` is used, but that would have to be vendored in - so you could either 

- keep the format the way you have it
- write out the file in json format and use the built-in json parser
- use yaml and vendor in `"github.com/ghodss/yaml"`

Reply:
Ok.  I'd go for "write out the file in json format and use the built-in json parser", which looks most light weight...